app/models/vat.rb

```diff
@@ -0,0 +1,33 @@
+module Shop
+  module Models
+    class VAT
+      class UnknownCategoryError < StandardError; end
+
+      class << self
+        private
+
+        def new(*args)
+          super(*args)
+        end
+
+        public
+
+        def for_category(id)
+          raise UnknownCategoryError unless @instances.key?(id)
+          @instances[id]
+        end
```

Comment:
if you will change order of those methods you will not need to use `public` definition